pkg/generate/app/cmd/describe.go

```diff
@@ -143,6 +143,9 @@ func describeBuildPipelineWithImage(out io.Writer, ref app.ComponentReference, p
 			fmt.Fprintf(out, "    * This image declares volumes and will default to use non-persistent, host-local storage.\n")
 			fmt.Fprintf(out, "      You can add persistent volumes later by running 'volume dc/%s --add ...'\n", pipeline.Deployment.Name)
 		}
+		if pipeline.Image.Info != nil && (len(pipeline.Image.Info.Config.User) == 0 || pipeline.Image.Info.Config.User == "root" || pipeline.Image.Info.Config.User == "0") {
```

Comment:
do we want to give you a separate warning text for len(user) == 0 (technically it's not the same as specifying user root)... and given that we'll run you with an arbitrary user, maybe it's what's preferred? 
Also if you do specify a specific user ... say 1001. Do we want to let you know that we won't run you with the user specified in your image?